src/tools/log-parser-rs/src/main.rs

```diff
@@ -0,0 +1,39 @@
+mod args;
+mod log_message;
+mod parse_file;
+
+use args::Cli;
+use clap::Parser;
+use log_message::LogMessage;
+use parse_file::*;
+use std::{path::PathBuf, process::exit};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum LogParserError {
+    #[error("Unkown Error")]
+    Unknown,
+
+    #[error("Input file '{0}' cannot be found!")]
+    InputFileNotFound(PathBuf),
+
+    #[error("Do not have permission to open '{0}'!")]
```

Comment:
Could you remove the exclamation marks from these errors for consistency with other Kata errors?